metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -2755,31 +2823,46 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         if self.server is None or not self.server.is_signed_in():
             return
         try:
-            # Initialise the dictionary to later look-up for chart and dashboard stat
-            if self.config.extract_usage_stats:
-                self._populate_usage_stat_registry()
-
-            self._populate_projects_registry()
-            yield from self.emit_project_containers()
-            yield from self.emit_workbooks()
-            if self.sheet_ids:
-                yield from self.emit_sheets()
-            if self.dashboard_ids:
-                yield from self.emit_dashboards()
-            if self.embedded_datasource_ids_being_used:
-                yield from self.emit_embedded_datasources()
-            if self.datasource_ids_being_used:
-                yield from self.emit_published_datasources()
-            if self.custom_sql_ids_being_used:
-                yield from self.emit_custom_sql_datasources()
-            if self.database_tables:
-                yield from self.emit_upstream_tables()
+            if self.config.ingest_multiple_sites:
+                for site in TSC.Pager(self.server.sites):
+                    if site.state != 'Active' or not self.config.site_name_pattern.allowed(site.name):
+                        continue
+                    self.current_site = site
+                    self.server.auth.switch_site(site)
```

Comment:
I just noticed, that the documentation states that this method is not allowed with Tableau Cloud: https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_authentication.htm#switch_site
Unfortunately, I'm not able to test it with Tableau Cloud. I had a solution in a previous commit that did the switch manually (by just re-authenticating). We could go back to this if needed. WDYT?